File: backend/src/main/java/com/example/ecomanager/dtos/ScoreDTO.java
package com.example.ecomanager.dtos;

import com.fasterxml.jackson.annotation.JsonProperty;
import jakarta.validation.constraints.Min;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@NoArgsConstructor
@AllArgsConstructor
@Getter
@Setter
@Builder
public class ScoreDTO {

    private Long employeeId;

    private Long meetingId;

    @NotNull(message = "Score is required!")
    @Min(value = 0, message = "Score must be non-negative!")
    private int score;

    private String comment;

}

File: frontend/src/app/dtos/score/score.dto.ts
import { IsNotEmpty, IsNumber } from 'class-validator';

export class ScoreDto {
  employeeId: number;

  meetingId: number;

  @IsNotEmpty()
  @IsNumber()
  score: number;

  comment?: string;

  constructor(data: any) {
    this.employeeId = data.employeeId;
    this.meetingId = data.meetingId;
    this.score = data.score;
    this.comment = data.comment;
  }
}
